packages/file_selector/file_selector_web/lib/file_selector_web.dart

```diff
@@ -174,9 +175,10 @@ class FileSelectorPlugin extends FileSelectorPlatform {
     });
 
     element.onError.first.then((event) {
-      if (!_completer.isCompleted) {
-        _completer.completeError(event);
-      }
+      throw PlatformException(
+          code: 'file_input',
+          message: "Input element failed on event with target: " +
+              event?.target?.toString());
     });
```

Comment:
@ditman Unsure what information to provide here. Does this look okay? Also not sure what to provide for code parameter

Reply:
Instead of throw, do:

```dart
_completer.completeError(PlatformException(...
```

if you then:

```
await getFilesWhenReady(...);
```

Whatever Error you've completed with will be thrown thrown for you. If people use the method as a traditional Future, they'll be able to catch the error manually.

Re: info to provide: try to extract information from the error event?

https://api.dart.dev/stable/2.9.2/dart-html/ErrorEvent-class.html

```dart
code: event.type,
message: event.message,
...
```